templates/user/edit-profile.html

```diff
@@ -177,6 +201,78 @@
+                if (confirm("{{ _('Are you sure you want to generate or regenerate a new set of scratch codes?') }}\n"
+                          + "{{ _('This will invalidate any previous scratch codes you have.') }} "
+                          + "{{ _('You will not be able to view your scratch codes after you leave this page!') }}")) {
+                    $('#scratch-codes').text("{{ _('Generating...') }}");
+
+                    var info_float = $('.info-float');
+                    if (info_float.length) {
+                        var container = $('#content-right');
+                        if (window.bad_browser) {
+                            container.css('float', 'right');
+                        } else if (!featureTest('position', 'sticky')) {
+                            fix_div(info_float, 55);
+                            $(window).resize(function () {
+                                info_float.width(container.width());
+                            });
+                            info_float.width(container.width());
+                        }
+                    }
+
+                    var copyButton;
+                    $('pre code').each(function () {
+                        $(this).parent().before($('<div>', {'class': 'copy-clipboard', 'id': {}})
+                            .append(copyButton = $('<span>', {
+                                'class': 'btn-clipboard',
+                                'id': 'scratch-codes-copy-button',
+                                'data-clipboard-text': '',
+                                'title': 'Click to copy'
+                            }).text('Copy')));
+
+                        $(copyButton.get(0)).mouseleave(function () {
+                            $(this).attr('class', 'btn-clipboard');
+                            $(this).removeAttr('aria-label');
+                        });
+
+                        var curClipboard = new Clipboard(copyButton.get(0));
+
+                        curClipboard.on('success', function (e) {
+                            e.clearSelection();
+                            showTooltip(e.trigger, 'Copied!');
+                        });
+
+                        curClipboard.on('error', function (e) {
+                            showTooltip(e.trigger, fallbackMessage(e.action));
+                        });
+                    });
+
+                    $.ajax({
+                        type: 'POST',
+                        url: $(this).attr('href'),
+                        dataType: 'json',
+                        success: function(data) {
+                            $('#scratch-codes').text(data.data.codes);
+                            $('#scratch-codes-copy-button').attr('data-clipboard-text', data.data.codes);
+                            $('#generate-scratch-codes-button').text("{{ _('Regenerate') }}");
+                            $('#scratch-codes-text').text("{{ _('Below is a list of one-time use scratch codes.') }} "
```

Comment:
Let's use a single `_('...')` block here too, so that we don't have to translate all these strings twice.